public/app/features/transformers/calculateHeatmap/heatmap.ts

```diff
@@ -208,13 +206,23 @@ export function calculateHeatmapFromData(frames: DataFrame[], options: HeatmapCa
         name: 'xMin',
         type: xField.type,
         values: new ArrayVector(heat2d.x),
-        config: xField.config,
+        config: {
+          ...xField.config,
+          custom: {
+            originalName: xField.name,
```

Comment:
```suggestion
            originalName: getFieldDisplayName( xField, ... ),
```
will include any fancy formatting etc